src/fides/api/ops/util/consent_util.py

```diff
@@ -34,9 +35,16 @@ def filter_privacy_preferences_for_propagation(
 
 def should_opt_in_to_service(
     system: Optional[System], privacy_request: PrivacyRequest
-) -> Optional[bool]:
+) -> Tuple[Optional[bool], List[PrivacyPreferenceHistory]]:
     """
-    Determine if we should opt in (return True), opt out (return False), or do nothing (return None) for the given System.
+    For SaaS Connectors, examine the Privacy Preferences and collapse this information into a single should we opt in? (True),
+    should we opt out? (False) or should we do nothing? (None).
+
+    Email connectors should instead call "filter_privacy_preferences_for_propagation" directly, since we can have preferences with
+    conflicting opt in/opt out values for those connector types.
+
+    Also return filtered preferences here so we can cache affected systems and/or secondary identifiers directly on these
+    filtered preferences for consent reporting.
```

Comment:
This method was just returning an optional boolean for saas connectors, as to whether we should opt in or out.  Now I'm also returning the subset of privacy preferences that are relevant for this service, so I can cache things like "affected system status" or "secondary user ids" for consent reporting purposes.